src/mlpack/methods/naive_bayes/nbc_main.cpp

```diff
@@ -188,17 +201,23 @@ static void mlpackMain()
     model->nbc.Classify(testingData, predictions, probabilities);
     Timer::Stop("nbc_testing");
 
-    if (CLI::HasParam("output"))
+    if (CLI::HasParam("output") || CLI::HasParam("predictions"))
     {
       // Un-normalize labels to prepare output.
       Row<size_t> rawResults;
       data::RevertLabels(predictions, model->mappings, rawResults);
-
-      // Output results.
-      CLI::GetParam<Row<size_t>>("output") = std::move(rawResults);
+      predictions = rawResults;
+      CLI::GetParam<Row<size_t>>("predictions") = predictions;
+      if (CLI::HasParam("output"))
+      {
+        CLI::GetParam<Row<size_t>>("output") = std::move(predictions);
+      }
+    }
+    CLI::GetParam<mat>("probabilities") = probabilities;
+    if (CLI::HasParam("output_probs"))
+    {
+      CLI::GetParam<mat>("output_probs") = probabilities;
```

Comment:
This code does not check if `CLI::HasParam("probabilities")` is true, and when you use `probabilities` the second time you can use `std::move()`.  Please read the documentation for `std::move()` (and move semantics) to understand why. :+1:

Reply:
Yea,that was a mistake on my part.